universe/Effect.h

```diff
@@ -1123,7 +1123,7 @@ class FO_COMMON_API GenerateSitRepMessage : public EffectBase {
     const std::string& Icon() const
     { return m_icon; }
 
-    const auto& MessageParameters() const
+    auto MessageParameters() -> const std::vector<std::pair<std::string, ValueRef::ValueRefBase<std::string>*>>&
     { return m_message_parameters; }
```

Comment:
@geoffthemedio `decltype(auto)` is a [C++14 feature](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3638.html).  Are you pushing us to migrate to C++14 ASAP 😉 

Does VC++ default to C++14?  We have explicitly restricted gcc to C++11 features in the CMakeList.txt.

Upgrading to C++14 requires no changes for our minimum required compiler version for VC++ (2015/ MSC_VER = 1900) and gcc (5.0). 

I can't tell what minimum version that we require for clang.  I think that upgrading to C++14 requires a change from clang 3.3 to 3.4. The release dates for those versions are 17 June 2013 and 2 January 2014 respectively.